presto/server.nim

```diff
@@ -0,0 +1,220 @@
+#
+#         REST API framework implementation
+#             (c) Copyright 2021-Present
+#         Status Research & Development GmbH
+#
+#              Licensed under either of
+#  Apache License, version 2.0, (LICENSE-APACHEv2)
+#              MIT license (LICENSE-MIT)
+import std/[options, json, strutils]
+import chronos, chronos/apps/http/httpserver
+import chronicles
+import stew/results
+import route, common, segpath, servercommon
+
+export options, chronos, httpserver, servercommon
+
+type
+  RestServer* = object of RootObj
+    server*: HttpServerRef
+    router*: RestRouter
+
+  RestServerRef* = ref RestServer
+
+proc getContentBody(r: HttpRequestRef): Future[Option[ContentBody]] {.async.} =
+  if r.meth notin PostMethods:
+    return none[ContentBody]()
+  else:
+    var default: seq[byte]
+    let cres = getContentType(r.headers.getList("content-type"))
+    if not(cres.isOk()):
+      raise newException(RestBadRequestError, "Incorrect Content-Type header")
+    let data =
+      if r.hasBody():
+        await r.getBody()
+      else:
+        default
+    let cbody = ContentBody(contentType: cres.get(), data: data)
+    return some[ContentBody](cbody)
+
+proc processRestRequest(server: RestServerRef,
```

Comment:
This code is too complicated to be duplicated in two files. How come it cannot be reused?